# Jane Street Training set correlations
[Kaggle Jane Street Market Prediction](https://www.kaggle.com/c/jane-street-market-prediction/overview)


In [6]:
import pandas as pd
import numpy as np
import datatable as dt
from datatable import f
tra = dt.fread('jane-street-market-prediction/train.csv')
#X = dt.fread('jane-street-market-prediction/train.csv', max_nrows=10000, skip_to_line=520000).to_pandas()

In [7]:
tra.shape

(2390491, 138)

In [10]:
tra.columns

AttributeError: 'datatable.Frame' object has no attribute 'columns'

In [8]:
del tra[f.date < 85, 'ts_id']

In [9]:
tra.shape

(2390491, 138)

In [5]:
corr = X.corr().resp.sort_values()

In [6]:
pd.set_option('display.max_rows', None)

In [7]:
corr

feature_17    -0.029358
feature_37    -0.027586
feature_21    -0.027208
feature_23    -0.024222
feature_0     -0.020868
feature_25    -0.018643
feature_38    -0.017517
feature_18    -0.015337
feature_22    -0.013907
feature_19    -0.013668
feature_24    -0.012732
feature_103   -0.010529
feature_51    -0.008315
feature_26    -0.008311
feature_97    -0.008253
feature_52    -0.007638
feature_106   -0.007427
feature_83    -0.006830
feature_50    -0.006810
feature_77    -0.006569
weight        -0.005711
feature_41    -0.005593
feature_20    -0.005239
feature_49    -0.004032
feature_105   -0.003803
feature_72    -0.003408
feature_100   -0.003249
feature_115   -0.002191
feature_121   -0.002111
feature_78    -0.002068
feature_117   -0.002035
feature_118   -0.001904
feature_92    -0.001838
feature_125   -0.001582
feature_48    -0.001499
feature_68    -0.001376
feature_45    -0.001363
feature_65    -0.001351
feature_67    -0.001109
date          -0.001072
feature_74    -0.001070
feature_80    -0

In [1]:
import pandas as pd
import numpy as np
import datatable as dt
tra = dt.fread('jane-street-market-prediction/train.csv')
#X = dt.fread('jane-street-market-prediction/train.csv', max_nrows=10000, skip_to_line=520000).to_pandas()

In [8]:
x27 = X.groupby('date').feature_27.median()
x27

date
86     0.079092
87     0.119160
88    -0.018156
89    -0.014805
90    -0.049628
91     0.141165
92    -0.062782
93    -0.016454
94     0.073848
95    -0.015151
96     0.071779
97     0.109461
98    -0.004316
99     0.194111
100   -0.017250
101    0.094849
102    0.128371
103   -0.025205
104   -0.025333
105    0.113037
106    0.164540
107    0.234815
108    0.106449
109    0.070464
110    0.213274
111   -0.045582
112   -0.045967
113   -0.001897
114   -0.009617
115    0.004588
116    0.089480
117    0.069390
118    0.051987
119    0.118687
120    0.002726
121   -0.099447
122    0.002026
123    0.084021
124    0.100959
125    0.074082
126    0.002688
127    0.007404
128   -0.015291
129   -0.066036
130   -0.028379
131   -0.042662
132    0.037513
133   -0.045439
134    0.072954
135   -0.077924
136   -0.036452
137   -0.014271
138   -0.019172
139    0.116047
140    0.050046
141   -0.045482
142   -0.101075
143   -0.033194
144   -0.037777
145   -0.045729
146   -0.058669
147   -0.011448
148

In [26]:
ind = [4, 5, 6, 7]
pd.DataFrame(['a', 'b', 'c', 'd'], index=ind)

0
4  a
5  b
6  c
7  d

In [21]:
len(pd.unique(X.date))

414

In [2]:
Xt = X[X.date >= 460]
yt = np.where((Xt.resp > 0), 1, 0)
X = X[(X.date > 85) & (X.date < 460)]
y = np.where((X.resp > 0), 1, 0)
drop_cols = ['date', 'resp_1', 'resp_2', 'resp_3', 'resp_4', 'resp', 'ts_id']
X.drop(drop_cols, axis=1, inplace=True)
Xt.drop(drop_cols, axis=1, inplace=True)

In [3]:
from sklearn.experimental import enable_hist_gradient_boosting
from sklearn.ensemble import HistGradientBoostingClassifier
clf = HistGradientBoostingClassifier(max_iter=100, verbose=0, random_state=42).fit(X, y)

In [4]:
clf.score(Xt, yt)

0.5203650466860196

### max_iter=100, verbose=0, random_state=42
0.5211998142876572
kaggle 5171.245

### drop feature_61
0.5215234039759324

### drop feature_61, 'feature_64'
0.5201024231708976

### drop feature_61, 'feature_90
0.5203650466860196